proxy/proxy.go

```diff
@@ -0,0 +1,71 @@
+package proxy
+
+import (
+	"net/http"
+	"sync"
+
+	"github.com/gorilla/websocket"
+	"github.com/taskcluster/webhooktunnel/util"
+	"github.com/taskcluster/webhooktunnel/wsmux"
+)
+
+type Config struct {
+	Upgrader websocket.Upgrader
+	Logger   util.Logger
+}
+
+type proxy struct {
+	m        sync.RWMutex
+	pool     map[string]*wsmux.Session
+	upgrader websocket.Upgrader
+	logger   util.Logger
+}
+
+func (p *proxy) GetHandler() http.Handler {
+	return http.HandlerFunc(p.handler)
+}
+
+func (p *proxy) validateRequest(r *http.Request) error {
+	return nil
+}
+
+func NewProxy(conf Config) *proxy {
```

Comment:
Exported function needs a documentation comment...


Also just rename the function `New()` since the package is `proxy` or `whproxy`, it becomes `whproxy.New(...)`